crates/sui-graphql-rpc/src/context_data/db_data_provider.rs

```diff
@@ -958,6 +966,60 @@ impl PgManager {
             Ok(None)
         }
     }
+    pub(crate) async fn fetch_staked_sui(
+        &self,
+        address: SuiAddress,
+        first: Option<u64>,
+        after: Option<String>,
+        last: Option<u64>,
+        before: Option<String>,
+    ) -> Result<Option<Connection<String, Stake>>, Error> {
+        let obj_filter = ObjectFilter {
+            package: None,
+            module: None,
+            ty: Some(MoveObjectType::staked_sui().to_string()),
+            owner: Some(address),
+            object_ids: None,
+            object_keys: None,
+        };
+
+        // we want the object ids of staked sui
+        let objs = self
+            .multi_get_objs(first, after, last, before, Some(obj_filter))
+            .await?;
+
+        let governance_api = GovernanceReadApiV2::new(self.inner.clone());
+
+        if let Some((stored_objs, has_next_page)) = objs {
+            let mut connection = Connection::new(false, has_next_page);
+
+            let object_ids = stored_objs
+                .into_iter()
+                .filter_map(|x| {
+                    ObjectID::try_from(x.object_id)
+                        .map_err(|_| eprintln!("Error extracting ObjectID from StoredObject"))
+                        .ok()
+                })
+                .collect::<Vec<_>>();
+            let stakes = governance_api.get_stakes_by_ids(object_ids).await?;
```

Comment:
This is unneeded, we can directly convert the fetched objects into staked sui

Reply:
We should directly use the get owned stakes api on governance api instead

Reply:
More context: since we are using governance api already, we can just use the method below:

async fn get_staked_by_owner(
        &self,
        owner: SuiAddress,
    ) -> Result<Vec<DelegatedStake>,

The graphQL::StakedSui is closest to DelegatedStake, but most of the fields can already be derived directly from the StoredObjecf. It's really just the estimated reward field that needs system state lookup.

So an alternative approach would be to directly convert stored object into stakedsui, and defer resolution of the estimated reward field 

However I think this can be done later and incurs more effort so let's just use get stakes by owner

Reply:
Hmm, I remember that was playing tricks on me. I can look again into it.

Reply:
I think the reason was that by getting the object ids, I can bypass having to deal with DelegatedStakes? Not sure I remember why, let me look into it.

Reply:
Wait, don't we want the DelegatedStake for the estimated reward? Otherwise there's no need to use governance api as the stored object already has staked sui info

Reply:
Right, for now we skip estimated reward due to some perf issues.